src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorMapToDocumentRangesEndpoint.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common.Extensions;
+using Microsoft.AspNetCore.Razor.LanguageServer.EndpointContracts;
+using Microsoft.AspNetCore.Razor.LanguageServer.Extensions;
+using Microsoft.AspNetCore.Razor.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class RazorMapToDocumentRangesEndpoint : IRazorMapToDocumentRangesHandler
+    {
+        private readonly RazorDocumentMappingService _documentMappingService;
+
+        public RazorMapToDocumentRangesEndpoint(RazorDocumentMappingService documentMappingService)
+        {
+            _documentMappingService = documentMappingService;
+        }
+
+        public bool MutatesSolutionState => false;
+
+        public TextDocumentIdentifier GetTextDocumentIdentifier(RazorMapToDocumentRangesParams request)
+        {
+            return new TextDocumentIdentifier
+            {
+                Uri = request.RazorDocumentUri
+            };
```

Comment:
General comment.. is there an advantage to us, in future, changing our params types to use a TextDocumentIdentifier instead of just a Uri?

Or conversely, doesn't CLaSP allow us to have GetTextDocumentIdentifier be typed as a Uri for us?

Reply:
If future versions of the LSP spec added other properties to TextDocumentIdentifier it would have use future-proofed, but I can modify RazorRequestContextFactory to accept Uri's for now too.